pywr/solvers/cython_glpk.pyx

```diff
@@ -104,6 +105,13 @@ cdef class CythonGLPKSolver:
         cdef int status
         cdef cross_domain_row
         cdef int n, num
+        cdef jmp_buf env_buffer
+        cdef int jmpval
+
+        glp_error_hook(error_hook, &env_buffer)
+        jmpval = setjmp(env_buffer)
+        if jmpval != 0:
+            raise GLPKError("An unexpected error was raised by GLPK")
```

Comment:
Do we need to call `glp_error_hook(NULL, NULL)` when we're finished? Maybe in `__dealloc__`?

Reply:
It occurs to me that this error hook must be global, rather than linked to a specific problem. What does it do when there are two models?

Reply:
Indeed. Seems we have a choice. A global hook for all GLPK solvers, etc. Or add and remove the hook as needed, c.f. my comment about a context manager.